src/encode/util.rs

```diff
@@ -1,37 +1,34 @@
-use std::hash;
 use std::io;
 
 // A Write computing a digest on the bytes written.
-pub struct HasherWrite<'a, W, H>
+pub struct DigestWrite<'a, 'b, W, S>
 where
     W: 'a + io::Write,
-    H: 'a + hash::Hasher,
+    S: crc::Width,
 {
-    write: &'a mut W,  // underlying writer
-    hasher: &'a mut H, // hasher
+    write: &'a mut W,                   // underlying writer
+    digest: &'a mut crc::Digest<'b, S>, // hasher
 }
 
-impl<'a, W, H> HasherWrite<'a, W, H>
+impl<'a, 'b, W, S> DigestWrite<'a, 'b, W, S>
 where
     W: io::Write,
-    H: hash::Hasher,
+    S: crc::Width,
 {
-    pub fn new(write: &'a mut W, hasher: &'a mut H) -> Self {
-        Self { write, hasher }
+    pub fn new(write: &'a mut W, digest: &'a mut crc::Digest<'b, S>) -> Self {
+        Self { write, digest }
     }
 }
 
-impl<'a, W, H> io::Write for HasherWrite<'a, W, H>
+impl<'a, 'b, W> io::Write for DigestWrite<'a, 'b, W, u32>
```

Comment:
Because of how [`impl Digest`](https://docs.rs/crc/latest/crc/struct.Digest.html) is defined there's no way to define it once for all possible types.

In practice we only need `u32` for `impl Write` for usage in lzma-rs, so it's only implemented for `u32` here.